### PR TITLE
Add an option to show violation shortlinks in the formatter output

### LINKED ISSUE #2374: Violation shortlinks in the formatter
I've set up a short URL for violations at https://pyflak.es/. It works like this:

- for WPS violations, it links to the specific violation in the WPS docs: https://pyflak.es/WPS235
- for violations present on Grant McConnaughey's [Flake8 Rules][1], it links to the respective page: https://pyflak.es/E117
- for other violations, it links back to the respective Flake8 plugin docs: https://pyflak.es/D300, https://pyflak.es/RST202

[1]: https://www.flake8rules.com/

It's currently semi-manually generated and a few links are broken (which I intend to fix soon-ish).

Would you be interested in integrating this deeper with wemake-python-styleguide? For example, the `wemake` formatter could print the URLs out, for easier documentation lookup.

### PR DESCRIPTION
**Pull request by notpushkin**
## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [ ] I have added my changes to the `CHANGELOG.md`

## Related issues

Closes #2374

### CODE REVIEW

**Comment by notpushkin** [comment]
@sobolevn I'm currently looking at how testing is done for the formatter. Should I just add a variation to the `@parametrize()` decorator in [test_formatter_output.py](https://github.com/notpushkin/wemake-python-styleguide/blob/master/tests/test_formatter/test_formatter_output.py) and commit the generated snapshots?

File: tests/test_formatter/snapshots/snap_test_formatter_output.py [hunk 1]
```diff
@@ -107,6 +107,23 @@
 https://wemake-python-stylegui.de/en/xx.xx/pages/usage/violations/
 '''
 
+snapshots['test_formatter[cli_options3-with_links] formatter_with_links'] = '''
+\x1b[4m\x1b[1m./tests/fixtures/formatter/formatter1.py\x1b[0m\x1b[0m
+  1:1      WPS111 Found too short name: s < 2 (https://pyflak.es/WPS111)
+  1:7      WPS110 Found wrong variable name: handle (https://pyflak.es/WPS110)
+  2:21     WPS432 Found magic number: 200 (https://pyflak.es/WPS432)
+  2:21     WPS303 Found underscored number: 2_00 (https://pyflak.es/WPS303)
```
Comment (sobolevn) [inline tests/test_formatter/snapshots/snap_test_formatter_output.py:115]: Another idea: we can print links on new lines. Why?

1. It maybe more readable and easier to click
2. Chars like `)` won't be considered as a part of the URL by some terminals (hypothetical)

File: wemake_python_styleguide/formatter.py [hunk 2]
```diff
@@ -83,11 +83,15 @@ def handle(self, error: Violation) -> None:  # noqa: WPS110
...
     def format(self, error: Violation) -> str:  # noqa: WPS125
         """Called to format each individual :term:`violation`."""
-        return '{newline}  {row_col:<8} {code:<5} {text}'.format(
+        return '{newline}  {row_col:<8} {code:<5} {text}{link}'.format(
             newline=self.newline if self._should_show_source(error) else '',
             code=error.code,
             text=error.text,
             row_col='{0}:{1}'.format(error.line_number, error.column_number),
+            link=(
+                ' (https://pyflak.es/{0})'.format(error.code)
```
Comment (sobolevn) [inline wemake_python_styleguide/formatter.py:92]: Let's move this into a constant at the top of this module.

**Review by sobolevn — COMMENTED** [review]
Thank you! 👍

File: tests/test_formatter/snapshots/snap_test_formatter_output.py (hunk 1 above)
Comment (notpushkin) [reply tests/test_formatter/snapshots/snap_test_formatter_output.py:115]: Something along the lines of:

```
  1:1      WPS111 Found too short name: s < 2 
           -> https://pyflak.es/WPS111
  1:7      WPS110 Found wrong variable name: handle (https://pyflak.es/WPS110)
           -> https://pyflak.es/WPS110
  2:21     WPS432 Found magic number: 200 (https://pyflak.es/WPS432)
           -> https://pyflak.es/WPS432
  2:21     WPS303 Found underscored number: 2_00 (https://pyflak.es/WPS303)
           -> https://pyflak.es/WPS303
```

?

Also, would it play well with `--show-source`?

Comment (sobolevn) [reply tests/test_formatter/snapshots/snap_test_formatter_output.py:115]: Let's add a test for it.

What do you think should go first?

Comment (notpushkin) [reply tests/test_formatter/snapshots/snap_test_formatter_output.py:115]: I've tried both styles and `line, source, link` looks a bit neater IMO:

```
wemake_python_styleguide/formatter.py

  98:77    C812  missing trailing comma
  row_col='{0}:{1}'.format(error.line_number, error.column_number)
                                                                  ^
  -> https://pyflak.es/C812

  121:1    D202  No blank lines allowed after function docstring
  """Called when ``--show-violation-links`` option is provided."""
  ^
  -> https://pyflak.es/D202
```

Implemented this one. For comparison:

<details><summary><code>line, link, source</code></summary>

```
wemake_python_styleguide/formatter.py

  98:77    C812  missing trailing comma
           -> https://pyflak.es/C812
  row_col='{0}:{1}'.format(error.line_number, error.column_number)
                                                                  ^

  121:1    D202  No blank lines allowed after function docstring
           -> https://pyflak.es/D202
  """Called when ``--show-violation-links`` option is provided."""
  ^
```

</details>

Comment (sobolevn) [reply tests/test_formatter/snapshots/snap_test_formatter_output.py]: I really like the second version. This way we have separate spacing for "meta" information and separate spacing for "source code".

Comment (notpushkin) [reply tests/test_formatter/snapshots/snap_test_formatter_output.py]: Can do.

**Comment by notpushkin** [comment]
How should we deal with `test_configuration`? I'm not sure if this option can be documented in the same way as violation options (I've documented it on the [Formatter][1] page though). Maybe just add an exception for it (`if option_item is not in FORMATTING_OPTIONS: assert`)?

[1]: https://wemake-python-styleguide--2375.org.readthedocs.build/en/2375/pages/usage/formatter.html

**Comment by notpushkin** [comment]
Also, any ideas how to get that last 1% of coverage for `wemake_python_styleguide/formatter.py`?

```
---------- coverage: platform darwin, python 3.10.2-final-0 ----------
Name                                    Stmts   Miss Branch BrPart  Cover   Missing
-----------------------------------------------------------------------------------
wemake_python_styleguide/formatter.py      84      0     24      1    99%   90->92
-----------------------------------------------------------------------------------
TOTAL                                    6393      0   2891      1    99%

141 files skipped due to complete coverage.
Coverage HTML written to dir htmlcov
Coverage XML written to file coverage.xml

FAIL Required test coverage of 100% not reached. Total coverage: 99.99%
```

**Comment by sobolevn** [comment]
I think that `line` is always present 🤔

File: docs/pages/usage/formatter.rst [hunk 3]
```diff
@@ -94,3 +94,23 @@ do you have and where you have them:
...
+.. code::
+
+  » flake8 . --format=wemake --show-source --show-violation-links
+
+  ./wemake_python_styleguide/formatter.py
+
+    E231  120:32   missing whitespace after ':'
+    def show_source(self, error:Violation) -> str:
+                               ^
+    -> https://pyflak.es/E231
```
Comment (sobolevn) [inline docs/pages/usage/formatter.rst:112]: This part looks outdated 🤔 
We now have a different format.

**Comment by sobolevn** [comment]
Now it should be fine! 👍

**Review by sobolevn — APPROVED** [review]
Thank you, great feature!

I also think that adding `--violation-links-prefix=` is a good idea.
So, users can change services:
- `--violation-links-prefix=https://ya.ru` will result in `https://ya.ru/WPS001`
- etc

If you are interested, you can send one more PR.

Bery big thanks for your work and financial support! 👍

**Pull request merged**